File: 0061-rotate-list/0061-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if(!head|| k==0) return head;
        ListNode* start = head;
        ListNode* ptr = head;
        int size = 1;
        while(start->next)
        {
            size++;
            start = start->next;
        }
        start->next= head;
        k = k%size;
        int steps = size - k;
        while(steps>1)
        {
            ptr = ptr->next;
            steps--;
        }
        head = ptr->next;
        ptr->next = NULL;
        return head;
    }
};

auto init = []
{
    ios::sync_with_stdio(NULL);
    cin.tie(NULL);
    cout.tie(NULL);
    return 'c';
}();

File: 0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if not head:
            return head
        size = 1
        ptr = head
        while(ptr.next):
            ptr = ptr.next
            size+=1
        k = k%size
        ptr.next = head
        ptr = head
        for _ in range( size - k - 1 ):
            ptr = ptr.next
        head = ptr.next
        ptr.next = None 
        return head
        